cmd/snap/cmd_info.go

```diff
@@ -0,0 +1,191 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"path/filepath"
+	"text/tabwriter"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/snap"
+)
+
+type infoCmd struct {
+	Verbose    bool `long:"verbose"`
+	Positional struct {
+		Snaps []string `positional-arg-name:"<snap>" required:"1"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+var shortInfoHelp = i18n.G("show detailed information about a snap")
+var longInfoHelp = i18n.G(`
+The info command shows detailed information about a snap, be it by name or by path.`)
+
+func init() {
+	addCommand("info",
+		shortInfoHelp,
+		longInfoHelp,
+		func() flags.Commander {
+			return &infoCmd{}
+		}, map[string]string{
+			"verbose": i18n.G("Include a verbose list of a snap's notes (otherwise, summarise notes)"),
+		}, nil)
+}
+
+func norm(path string) string {
+	path = filepath.Clean(path)
+	if osutil.IsDirectory(path) {
+		path = path + "/"
+	}
+
+	return path
+}
+
+func tryDirect(w io.Writer, path string, verbose bool) bool {
+	path = norm(path)
+
+	snapf, err := snap.Open(path)
+	if err != nil {
+		return false
+	}
+
+	info, err := snap.ReadInfoFromSnapFile(snapf, nil)
+	if err != nil {
+		return false
+	}
+	fmt.Fprintf(w, "path:\t%q\n", path)
+	fmt.Fprintf(w, "name:\t%s\n", info.Name())
+	fmt.Fprintf(w, "summary:\t%q\n", info.Summary())
+
+	var notes *Notes
+	if verbose {
+		fmt.Fprintln(w, "notes:\t")
+		fmt.Fprintf(w, "  confinement:\t%s\n", info.Confinement)
+		if info.Broken == "" {
+			fmt.Fprintln(w, "  broken:\tfalse")
+		} else {
+			fmt.Fprintf(w, "  broken:\ttrue (%s)\n", info.Broken)
+		}
+
+	} else {
+		notes = NotesFromInfo(info)
+	}
+	fmt.Fprintf(w, "version:\t%s %s\n", info.Version, notes)
+	if info.Type != snap.TypeApp {
+		fmt.Fprintf(w, "type:\t%s\n", info.Type)
```

Comment:
If this goes into 2.18 tomorrow, let's please be careful not to surface "os" in those strings.